wd_helper.au3

```diff
@@ -1035,18 +1035,18 @@ Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $aParameters
 						$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
 						$iErr = @error
 					EndIf
-				Case 'options'
+				Case 'options' ; 6 columns (value, label, index, selected status, disabled status, and hidden status)
 					$sScript = _
 							"var result ='';" & _
 							"var o = arguments[0].options;" & _
 							"for ( let i = 0; i < o.length; i++ )" & _
-							"  {result += o[i].value + '|' + o[i].label + '|' + o[i].index + '|' + o[i].selected  + '|' + (o[i].disabled || (o[i].parentNode.nodeName =='OPTGROUP' && o[i].parentNode.disabled)) + '\n'};" & _
+							"  {result += o[i].value + '|' + o[i].label + '|' + o[i].index + '|' + o[i].selected  + '|' + (o[i].disabled || (o[i].parentNode.nodeName =='OPTGROUP' && o[i].parentNode.disabled)) + '|' + (o[i].hidden || (o[i].parentNode.nodeName =='OPTGROUP' && o[i].parentNode.hidden))  + '\n'};" & _
```

Comment:
small refactoring proposal to keep the code simialar for all cases ( `multiSelect` and `options` should use similar JS code)
```autoit
				Case 'options' ; 6 columns (value, label, index, selected status, disabled status, and hidden status)
					$sScript = _
							"var result ='';" & _
							"for ( var i = 0, l = arguments[0].options.length, o; i < l; i++ )" & _
							"{" & _
							"  o = arguments[0].options[i];" & _
							"  result += o.value + '|' + o.label + '|' + o.index + '|' + o.selected  + '|' + (o.disabled || (o.parentNode.nodeName =='OPTGROUP' && o.parentNode.disabled)) + '|' + (o.hidden || (o.parentNode.nodeName =='OPTGROUP' && o.parentNode.hidden))  + '\n';" & _
							"};" & _
							"return result;"
```

Reply:
if you agree with such aproach , then similar change should be also taken for `selectAll`

Reply:
Basically your changes are good, I'm just wondering to standardize all the JavaScripts in this function.

Reply:
I see your point. However, I would propose that `multiSelect` should be updated so that it is more in line with the existing coding in `options` and `selectAll`.

Reply:
here is my fixed proposal:
```autoit
Case 'options' ; 6 columns (value, label, index, selected status, disabled status, and hidden status)
	$sScript = _
			"var result ='';" & _
			"var options = arguments[0].options;" & _
			"for (let i = 0, o; i < options.length; i++)" & _
			" {" & _
			"  o = options[i];" & _
			"  result += o.value + '|' + o.label + '|' + o.index + '|' + o.selected  + '|' + (o.disabled || (o.parentNode.nodeName =='OPTGROUP' && o.parentNode.disabled)) + '|' + (o.hidden || (o.parentNode.nodeName =='OPTGROUP' && o.parentNode.hidden))  + '\n'" & _
			" }" & _
			"return result;"

```

Reply:
```autoit
Case 'multiSelect' ; https://stackoverflow.com/a/1296068/5314940
	If UBound($aParameters, $UBOUND_DIMENSIONS) <> 1 Or UBound($aParameters, $UBOUND_ROWS) = 0 Then ; should be a single dimensional non-empty array
		$iErr = $_WD_ERROR_InvalidArgue
		$iExt = 41 ; $iExt from 41 to 49 are related to _WD_ElementSelectAction()
	Else
		$sScript = _
				"var LabelsToSelect = ['" & _ArrayToString($aParameters, "', '") & "'];" & _
				"var options = arguments[0].options;" & _
				"for ( var i = 0, o; i < options.length; i++ )" & _
				"{" & _
				"  o = options[i];" & _
				"  if ( ( LabelsToSelect.indexOf(o.label) != -1 ) && (o.disabled==false && (!(o.parentNode.nodeName =='OPTGROUP' && o.parentNode.disabled))) && (o.hidden==false && (!(o.parentNode.nodeName =='OPTGROUP' && o.parentNode.hidden))) )" & _
				"  {" & _
				"    o.selected = true;" & _
				"  }" & _
				"};" & _
				"arguments[0].dispatchEvent(new Event('change', {bubbles: true}));" & _
				"return true;"
		$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
		$iErr = @error
	EndIf
````
and
```autoit
Case 'selectAll'
	$sScript = _
			"var options = arguments[0].options;" & _
			"for ( i=0, o; i < options.length; i++)" & _
			" {" & _
			"  o = options[i];" & _
			"  if ( (o.disabled==false && (!(options.item(i).parentNode.nodeName =='OPTGROUP' && options.item(i).parentNode.disabled))) && (o.hidden==false && (!(options.item(i).parentNode.nodeName =='OPTGROUP' && options.item(i).parentNode.hidden))) )" & _
			"    {o.selected = true}" & _
			" }" & _
			"arguments[0].dispatchEvent(new Event('change', {bubbles: true}));" & _
			"return true;"
```
and
```autoit
Case 'selectedOptions' ; 4 columns (value, label, index and selected status)
	$sScript = _
			"var result ='';" & _
			"var options = arguments[0].selectedOptions;" & _
			"for (let i = 0; i < options.length; i++)" & _
			" {result += options[i].value + '|' + options[i].label + '|' + options[i].index + '|' + options[i].selected + '\n'};" & _
			"return result;"
```

Reply:
Now they show the same way to write JavaScript code.
Is this in line with your approach?

Reply:
above posted code was refactored

Reply:
I guess that should work (although you are missing a `let` on selectAll code). Please retest each of these options due to the code refactoring **before** submitting changes to PR. 😉

Reply:
Thanks for findings `let` issue.

I will test and then provide PR to yours https://github.com/Danp2/au3WebDriver/tree/dp_hidden

of course ASAP.

Reply:
This has been "detached" into it's own separate PR (#350)